ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java

```diff
@@ -13194,16 +13194,16 @@ private void updateDefaultTblProps(Map<String, String> source, Map<String, Strin
       }
     }
 
-    boolean makeInsertOnly = !isTemporaryTable && (isManaged && HiveConf.getBoolVar(
-        conf, ConfVars.HIVE_CREATE_TABLES_AS_INSERT_ONLY));
-    boolean makeAcid = !isTemporaryTable && !makeInsertOnly && makeAcid();
+    boolean makeInsertOnly = !isTemporaryTable && HiveConf.getBoolVar(
+        conf, ConfVars.HIVE_CREATE_TABLES_AS_INSERT_ONLY);
+    boolean makeAcid = !isTemporaryTable && makeAcid();
     // if not specify managed table and create.table.as.external is true
     // ignore makeInsertOnly and makeAcid.
     if (!isManaged && HiveConf.getBoolVar(conf, ConfVars.CREATE_TABLE_AS_EXTERNAL)) {
       makeInsertOnly = false;
       makeAcid = false;
     }
-    if ((makeInsertOnly || makeAcid || isTransactional)
+    if ((makeInsertOnly || makeAcid || isTransactional || isManaged)
         && !isExt  && !isMaterialization && StringUtils.isBlank(storageFormat.getStorageHandler())
```

Comment:
I do not understand the logic here: Why isManaged and without explicitly saying it is a transactional table, it is an insert-only one?

Reply:
multiple ways to create acid table
create transactional table
create managed table
create table with hive.create.as.acid=true or hive.create.as.insert.only=true.

so the above logic ensures that table is ACID table when managed keyword if used.